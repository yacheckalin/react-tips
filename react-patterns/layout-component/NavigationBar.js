import React from "react";

const NavigationBar = ({ search }) => (
  <nav>
    <div className="nav-wrapper">
      <a className="brand-logo right">Logo</a>
      {search}
    </div>
  </nav>
);

export default NavigationBar;
