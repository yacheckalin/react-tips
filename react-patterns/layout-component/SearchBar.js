import React from "react";

const SearchBar = () => (
  <form>
    <div className="input-field">
      <input id="search" type="search" required />
      <label htmlFor="search">
        <i className="material-icons">search</i>
      </label>
    </div>
  </form>
);

export default SearchBar;
