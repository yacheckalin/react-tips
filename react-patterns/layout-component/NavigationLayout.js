import React from "react";
import NavigationBar from "./NavigationBar";
import SearchBar from "./SearchBar";

const NavigationLayout = () => <NavigationBar search={<SearchBar />} />;

export default NavigationLayout;
