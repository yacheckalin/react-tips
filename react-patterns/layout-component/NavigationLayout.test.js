import React from "react";
import toJSON from "enzyme-to-json";
import NavigationLayout from "./NavigationLayout";
import NavigationBar from "./NavigationBar";
import SearchBar from "./SearchBar";
import { shallow } from "enzyme";

describe("<NavigationLayout />", () => {
  it("renders and displays properly", () => {
    const wrapper = shallow(<NavigationLayout />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it("renders and displays properly", () => {
    const wrapper = shallow(<NavigationBar />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it("renders and displays properly", () => {
    const wrapper = shallow(<SearchBar />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
