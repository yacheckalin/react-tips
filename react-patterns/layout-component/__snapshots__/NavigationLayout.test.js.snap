// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationLayout /> renders and displays properly 1`] = `
<NavigationBar
  search={<SearchBar />}
/>
`;

exports[`<NavigationLayout /> renders and displays properly 2`] = `
<nav>
  <div
    className="nav-wrapper"
  >
    <a
      className="brand-logo right"
    >
      Logo
    </a>
  </div>
</nav>
`;

exports[`<NavigationLayout /> renders and displays properly 3`] = `
<form>
  <div
    className="input-field"
  >
    <input
      id="search"
      required={true}
      type="search"
    />
    <label
      htmlFor="search"
    >
      <i
        className="material-icons"
      >
        search
      </i>
    </label>
  </div>
</form>
`;
